llvm/lib/Target/EraVM/EraVMTargetMachine.cpp

```diff
@@ -208,6 +209,7 @@ class EraVMPassConfig : public TargetPassConfig {
 
   void addIRPasses() override;
   bool addInstSelector() override;
+  bool addILPOpts() override;
```

Comment:
Since we need to run this pass before `MachineCSE`, we could either add it in `addMachineSSAOptimization` (and call base implementation after that) or in a `addILPOpts` hook that is called just before MachineLICM and MachineCSE.
The only reason why I decided to run it as an ILP optimization is because `EarlyTailDuplicate` is called before that, and if we want to implement same optimization as in `AArch64ConditionOptimizer` in a future, it will benefit from it since it can catch more cases.